### PR TITLE
Eng 25 wallet connect cleanup

### PR DESCRIPTION
**Pull request by artmanque**

### CODE REVIEW

File: packages/OreIdWalletConnect/src/OreIDWalletConnect/OreIDWalletConnect.tsx [hunk 1]
```diff
@@ -40,7 +40,11 @@ export const OreIDWalletConnect: React.FC<OreIDWalletConnectProps> = ({
     { peerMeta: PeerMeta; request: WalletConnectTransaction } | undefined
   >()
 
-  const hasChainNetworkSupport = hasChainSupport(config.chainNetwork)
+  if (!config.clientIcons || !config.clientDescription || !config.clientName) {
+    throw Error(`config missing at least one: clientIcons, clientDescription, clientName, clientUrl`)
+  }
+
+  const hasChainNetworkSupport = hasChainSupport(config?.chainNetwork)
```
Comment (yardz) [inline packages/OreIdWalletConnect/src/OreIDWalletConnect/OreIDWalletConnect.tsx:46]: `config.chainNetwork` should be mandatory.

File: packages/OreIdWalletConnect/src/ActiveSessionButton/ActiveSessionButton.scss [hunk 2]
```diff
@@ -7,6 +7,7 @@
   background: transparent;
   cursor: pointer;
   padding: 10px 12px;
+  white-space: nowrap;
```
Comment (yardz) [inline packages/OreIdWalletConnect/src/ActiveSessionButton/ActiveSessionButton.scss:10]: I believe this style should stay with the "parent" or make the button "inline-block"

File: packages/OreIdWalletConnect/src/ConnectionsBadge/ConnectionsBadge.tsx [hunk 3]
```diff
@@ -16,9 +16,9 @@ export const ConnectionsBadge: React.FC<ConnectedUIProps> = ({ isListening, peer
   return (
     <div className="oreIdWalletConnect-connectionsBadge">
       <div className="oreIdWalletConnect-connectionsBadge-icons">
-        {peerMeta.map((metaData) => {
+        {peerMeta.map((metaData, index) => {
           if (!metaData?.icons?.[0]) return null
-          return <ConnectionIcon key={metaData.url} icon={metaData.icons[0]} size={20} />
```
Comment (yardz) [inline packages/OreIdWalletConnect/src/ConnectionsBadge/ConnectionsBadge.tsx:21]: Don't use the array index with key if you have another option, it generates bugs... If you delete an item in the middle of the array you will have inconsistency problems to render.
only use index if you don't have another more suitable field or if you're sure what you're doing.

File: packages/OreIdWalletConnect/src/ConnectWalletWidget/ConnectWalletWidget.tsx [hunk 4]
```diff
@@ -37,9 +37,7 @@ export const ConnectWalletWidget: React.FC<ConnectWalletWidgetProps> = ({ onClic
         <div>
           <QrCodeImage size={30} />
         </div>
-        <div>
```
Comment (yardz) [inline packages/OreIdWalletConnect/src/ConnectWalletWidget/ConnectWalletWidget.tsx:40]: This div has a reason... Leave the spacing equal, as shown in the figma. And also ensure that CSS will not influence the internal component.

File: packages/OreIdWalletConnect/src/OreIDWalletConnect/OreIDWalletConnect.tsx [hunk 5]
```diff
@@ -83,8 +83,7 @@ export const OreIDWalletConnect: React.FC<OreIDWalletConnectProps> = ({
   const onConnectionCreate: WalletConnectRefEvent = (connection, payload) => {
     if (props.onConnectionCreate) {
       try {
-        const currentConnection = getCurrentConnectionByUri(connection.connector.uri)
```
Comment (yardz) [inline packages/OreIdWalletConnect/src/OreIDWalletConnect/OreIDWalletConnect.tsx:82]: Don't do this, you will get an error here. The variable received in connection may be out of date. The updated connection is always/only the one in useRef.

This code will insert bugs.

**Review by yardz — CHANGES_REQUESTED** [review]

File: packages/OreIdWalletConnect/src/ConnectWalletWidget/ConnectWalletWidget.tsx (hunk 4 above)
Comment (artmanque) [reply packages/OreIdWalletConnect/src/ConnectWalletWidget/ConnectWalletWidget.tsx:40]: its correct as per updated styles

File: packages/OreIdWalletConnect/src/ActiveSessionButton/ActiveSessionButton.scss (hunk 2 above)
Comment (artmanque) [reply packages/OreIdWalletConnect/src/ActiveSessionButton/ActiveSessionButton.scss:10]: i disagree

File: packages/OreIdWalletConnect/src/OreIDWalletConnect/OreIDWalletConnect.tsx (hunk 1 above)
Comment (artmanque) [reply packages/OreIdWalletConnect/src/OreIDWalletConnect/OreIDWalletConnect.tsx:46]: it is. why do you think its not?

File: packages/OreIdWalletConnect/src/ConnectionsBadge/ConnectionsBadge.tsx (hunk 3 above)
Comment (artmanque) [reply packages/OreIdWalletConnect/src/ConnectionsBadge/ConnectionsBadge.tsx:21]: there is no suitable field currently

**Pull request merged**